Yoast/yoast-components
# Implement the requests to work with the (WP)REST-API

This PR fixes a couple of things so that the wizard works correctly when loaded as a dependency in wordpress-seo. 

This PR fixes the following things:
- When saving the options, a PUT requests is send instead of a POST request.
- The JSON enconding is now correct, previously this was double encoded.
- The PUT requests now contain headers which are needed to put the WP nonce in the headers.
- The componentName in the config is renamed to the old component, because the WP-API expects this.

Also incorrect default props in the OnboardingWizard are fixed.

Testing:
1. Run grunt build and the npm start.
2. Go through the steps and see if everything goes ok and no errors are shown in the console.
3. Check if a PUT request is send by the wizard and the params for the request are JSON.
4. By default jQuery is not present, to test the jQuery functionalities add ```import jQuery from "jquery";``` to the top of postJSON.js. If jquery is not yet installed run: ``npm install jquery``
5. Now check if a put request is send when you go to the next or previous step. This request has to contain: ```"X-WP-Nonce": "test1234"``` in the headers.
